driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```diff
@@ -52,16 +52,16 @@
       domain equal to the domain of the srvHost.
     */
     @Override
-    public List<String> resolveHostFromSrvRecords(final String srvHost) {
+    public List<String> resolveHostFromSrvRecords(final String srvHost, final String srvServiceName) {
         String srvHostDomain = srvHost.substring(srvHost.indexOf('.') + 1);
         List<String> srvHostDomainParts = asList(srvHostDomain.split("\\."));
         List<String> hosts = new ArrayList<String>();
         InitialDirContext dirContext = createDnsDirContext();
         try {
-            Attributes attributes = dirContext.getAttributes("_mongodb._tcp." + srvHost, new String[]{"SRV"});
+            Attributes attributes = dirContext.getAttributes("_" + srvServiceName + "._tcp." + srvHost, new String[]{"SRV"});
             Attribute attribute = attributes.get("SRV");
             if (attribute == null) {
-                throw new MongoConfigurationException("No SRV records available for host " + srvHost);
+                throw new MongoConfigurationException("_" + "No SRV records available for " + "_" + srvServiceName + "._tcp." + srvHost);
```

Comment:
Based on the [resource record (RR) format](https://www.rfc-editor.org/rfc/rfc1035#section-3.2.1), this triplet is the part of a resource record called "name", which matches its meaning from the perspective of the `DirContext.getAttributes` method. So it is definitely correct to call the triplet `resourceRecordName`<sup>1</sup>.

---
<sup>1</sup> The "resourceRecord" prefix is useful to avoid the ambiguity caused by the fact that RRs of the [`type`](https://www.rfc-editor.org/rfc/rfc1035#section-3.2.2) 33 (`SRV`) introduce structure to the `name` (both the type value and the structure are specified in [RFC2782](https://www.rfc-editor.org/rfc/rfc2782)), and call one of its parts "name", which overloads the term.

Reply:
Done